proposals/bip-001.md

```diff
@@ -0,0 +1,107 @@
+# BIP Purpose and Guidelines
+
+| Field | Value |
+| --- | --- |
+| BIP | 1 |
+| Title | BIP Purpose and Guidelines |
+| Author | Alfredo Beaumont |
+| Status | Draft |
+| Created | 2017-05-26 |
+| Updated | 2017-05-26 |
+
+## Abstract
+
+BIP stands for Babelfish Improvement Proposal. A BIP is a design document which
+describes a change proposal to Babelfish. It's initially designed specifically
+to propose changes to [UAST](https://doc.bblf.sh/uast/specification.html)
+components regarding code categorization through Roles or related entities, but
+its scope may become broader in the future. It's informally defined.
```

Comment:
I don't think we should use BIPs for every change in Babelfish, only for those which require some deeper background and discussion. Features in utils/ are just a show case and their changes shouldn't be formalised. API changes that break backward compatibility may be included in BIPs, but I'd wait till the API is stabilized before formalising its changes.